### PR TITLE
bug[rust]: Fix float display

### LINKED ISSUE #4837: Float printing for display is surprising
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of polars.

### Issue Description

Float printing for a monotonically increasing series does not look monotonic. I pulled the example below from a cumulative histogram.

### Reproducible Example

```python
import polars as pl
B = 1004500
S = 1004590
pl.DataFrame({"a": [i * 5 + B for i in range(1, 5)]}).select(
    [pl.col("a"), (pl.col("a") / S).alias("b"), (1 - (pl.col("a") / S)).alias("c"), ]
)
Out [3]:
shape: (4, 3)
┌─────────┬──────────┬──────────┐
│ a       ┆ b        ┆ c        │
│ ---     ┆ ---      ┆ ---      │
│ i64     ┆ f64      ┆ f64      │
╞═════════╪══════════╪══════════╡
│ 1004505 ┆ 0.999915 ┆ 0.000085 │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
│ 1004510 ┆ 0.9999   ┆ 0.0      │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
│ 1004515 ┆ 0.999925 ┆ 0.000075 │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
│ 1004520 ┆ 0.9999   ┆ 0.0      │
└─────────┴──────────┴──────────┘
```

### Expected Behavior

Column "c" should render the values for the 2nd and 4th rows.

### Installed Versions

<details>
```  
pr.show_versions()
---Version info---
Polars: 0.14.5
Index type: UInt32
Platform: Linux-4.19.0-21-cloud-amd64-x86_64-with-glibc2.28
Python: 3.9.9 (main, Aug 25 2022, 19:32:03) 
[GCC 12.1.0]
---Optional dependencies---
pyarrow: 9.0.0
pandas: 1.4.3
numpy: 1.23.2
fsspec: 2022.7.1
connectorx: <not installed>
xlsx2csv: <not installed>
```
</details>

### PR DESCRIPTION
**Pull request by owrior**
Closes #4837 

- Moves iterator to after string reduction.

### CODE REVIEW

**Comment by owrior** [comment]
```
import polars as pl
B = 1004500
S = 1004590
pl.DataFrame({"a": [i * 5 + B for i in range(1, 5)]}).select(
    [pl.col("a"), (pl.col("a") / S).alias("b"), (1 - (pl.col("a") / S)).alias("c"), ]
)
Out [3]:
shape: (4, 3)
┌─────────┬──────────┬──────────┐
│ a       ┆ b        ┆ c        │
│ ---     ┆ ---      ┆ ---      │
│ i64     ┆ f64      ┆ f64      │
╞═════════╪══════════╪══════════╡
│ 1004505 ┆ 0.999915 ┆ 0.000085 │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
│ 1004510 ┆ 0.9999   ┆ 0.000080      │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
│ 1004515 ┆ 0.999925 ┆ 0.000075 │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
│ 1004520 ┆ 0.9999   ┆ 0.000070      │
└─────────┴──────────┴──────────┘
```

File: polars/polars-core/src/fmt.rs [hunk 1]
```diff
@@ -902,6 +902,24 @@ Series: 'foo' [i32]
 	97
 	98
 	99
+]"#,
+            format!("{:?}", s)
+        );
+    }
+
+    #[test]
```
Comment (ritchie46) [inline polars/polars-core/src/fmt.rs:910]: Can you hoist this test to the python library? We want to reduce compilation times of the rust tests.

**Review by ritchie46 — COMMENTED** [review]
Thanks @owrior. I have one remark about a test and then it is good to go.

File: polars/polars-core/src/fmt.rs (hunk 1 above)
Comment (owrior) [reply polars/polars-core/src/fmt.rs:910]: Ahh yes sure thing.

Comment (owrior) [reply polars/polars-core/src/fmt.rs]: @ritchie46 I think we are good now if this looks ok to you.

**Comment by ritchie46** [comment]
Thanks!

**Pull request merged**